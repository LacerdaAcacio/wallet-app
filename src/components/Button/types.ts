import { TouchableOpacityProps } from 'react-native';

export type ButtonVariant = 'primary' | 'secondary';

export interface ButtonProps extends TouchableOpacityProps {
  title: string;
  variant?: ButtonVariant;
  testID?: string;
}
