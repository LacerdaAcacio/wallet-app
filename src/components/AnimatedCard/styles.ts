import styled from 'styled-components/native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

interface CardStyleProps {
  variant?: 'black' | 'green';
}

export const CardWrapper = styled.View`
  margin-bottom: -${hp('20%')}px;
  align-items: center;
  width: 100%;
`;

export const CardContainer = styled.View<CardStyleProps>`
  width: ${wp('85%')}px;
  height: ${hp('25%')}px;
  background-color: ${({ theme, variant }) =>
    variant === 'black' ? theme.colors.black : theme.colors.secondaryButton};
  border-radius: ${({ theme }) => theme.radii.large};
  padding: ${wp('5%')}px;
  justify-content: space-between;
  elevation: 5;
  shadow-color: #000;
  shadow-offset: 0px 4px;
  shadow-opacity: 0.3;
  shadow-radius: 5px;
`;

const BaseText = styled.Text<CardStyleProps>`
  color: ${({ theme, variant }) =>
    variant === 'black' ? theme.colors.white : theme.colors.textOnSecondary};
  font-family: ${({ theme }) => theme.fonts.body};
`;

export const CardType = styled(BaseText)`
  font-size: ${({ theme }) => theme.fontSizes.h5};
  font-weight: bold;
`;

export const CardHolder = styled(BaseText)`
  font-size: ${({ theme }) => theme.fontSizes.body};
  margin-top: ${hp('2%')}px;
`;

export const CardNumberContainer = styled.View`
  margin-top: ${hp('1%')}px;
`;

export const CardNumberText = styled(BaseText)`
  font-size: ${({ theme }) => theme.fontSizes.body};
  letter-spacing: 2px;
`;

export const CardInfoRow = styled.View`
  flex-direction: row;
  justify-content: space-between;
  margin-top: ${hp('2%')}px;
`;

export const CardLabel = styled(BaseText)`
  font-size: ${({ theme }) => theme.fontSizes.small};
`;
