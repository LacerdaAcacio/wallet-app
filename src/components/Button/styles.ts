import styled from 'styled-components/native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

interface ButtonProps {
  variant?: 'primary' | 'secondary';
}

export const StyledButton = styled.TouchableOpacity<ButtonProps>`
  width: ${wp('85%')}px;
  max-width: 340px;
  height: ${hp('7%')}px;
  max-height: 60px;

  background-color: ${({ theme, variant }) =>
    variant === 'secondary' ? theme.colors.secondaryButton : theme.colors.primaryButton};
  border-radius: ${({ theme }) => theme.radii.medium};
  justify-content: center;
  align-items: center;
  margin-bottom: ${hp('2.5%')}px;
`;

export const ButtonText = styled.Text<ButtonProps>`
  color: ${({ theme, variant }) =>
    variant === 'secondary' ? theme.colors.textOnSecondary : theme.colors.textOnPrimary};
  font-size: ${wp('4.5%')}px;
`;
