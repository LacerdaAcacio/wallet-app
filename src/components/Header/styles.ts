import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from 'react-native-responsive-screen';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import styled from 'styled-components/native';

const shadowStyles = {
  elevation: 8,
  shadowColor: '#000',
  shadowOffset: { width: 0, height: 4 },
  shadowOpacity: 0.1,
  shadowRadius: 5,
};
interface SectionProps {
  variant: 'default' | 'wallet';
}

export const Container = styled.View`
  background-color: transparent;
`;

export const TopSection = styled.View.attrs<SectionProps>(props => ({
  style: props.variant === 'wallet' ? shadowStyles : {},
}))<SectionProps>`
  background-color: ${({ theme, variant }) =>
    variant === 'wallet' ? theme.colors.lightGray : 'transparent'};

  padding-top: ${getStatusBarHeight()}px;
  padding-bottom: ${hp('2%')}px;
  padding-horizontal: ${wp('5%')}px;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  z-index: 10;
`;
interface HeaderTitleProps {
  variant: 'default' | 'wallet';
}

export const HeaderTitle = styled.Text<HeaderTitleProps>`
  flex: 1;
  font-size: ${wp('7.5%')}px;
  text-align: center;

  color: ${({ theme, variant }) => {
    return variant === 'wallet' ? theme.colors.backgroundDark : theme.colors.primaryButton;
  }};
`;

export const BackButton = styled.TouchableOpacity`
  padding: ${wp('2%')}px;
`;

export const HeaderPlaceholder = styled.View`
  width: ${wp('3%')}px;
`;

export const SubHeaderSection = styled.View`
  width: 100%;
  padding-vertical: 20px;
  padding-horizontal: ${wp('5%')}px;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.lightGray};
  border-bottom-left-radius: 30px;
  border-bottom-right-radius: 30px;

  margin-top: -30px;
  padding-top: 40px;
`;

export const SubHeaderTitle = styled.Text`
  font-size: ${({ theme }) => theme.fontSizes.h3};
  color: ${({ theme }) => theme.colors.primaryButton};
`;

export const AddButton = styled.TouchableOpacity`
  position: absolute;
  right: ${wp('5%')}px;
  padding: 8px;
`;
