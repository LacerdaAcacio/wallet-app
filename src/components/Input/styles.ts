import styled from 'styled-components/native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

export const InputContainer = styled.View`
  width: ${wp('90%')}px;
  margin-bottom: ${hp('2%')}px;
`;

export const Label = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  font-size: ${wp('4%')}px;
  margin-bottom: ${hp('1%')}px;
`;

export const TextInputStyled = styled.TextInput`
  flex: 1;
  height: 100%;
  font-size: ${wp('4%')}px;
  color: ${({ theme }) => theme.colors.black};
  margin-left: ${wp('2%')}px;
`;

export const TextInputContainer = styled.View`
  background-color: ${({ theme }) => theme.colors.white};
  border-radius: ${({ theme }) => theme.radii.small};
  height: ${hp('6.5%')}px;
  padding-horizontal: ${wp('4%')}px;
  flex-direction: row;
  align-items: center;

  shadow-color: #000;
  shadow-offset: 0px 1px;
  shadow-opacity: 0.15;
  shadow-radius: 4px;
  elevation: 3;
`;

export const ErrorText = styled.Text`
  color: ${({ theme }) => theme.colors.error};
  font-size: ${wp('3.5%')}px;
  margin-top: ${hp('0.5%')}px;
  margin-left: ${wp('1%')}px;
`;
