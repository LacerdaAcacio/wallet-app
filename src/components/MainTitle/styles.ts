import styled from 'styled-components/native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

export const TitleText = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  font-size: ${wp('10%')}px;
  text-align: center;
  margin-bottom: ${hp('5%')}px;
  font-weight: bold;
`;
