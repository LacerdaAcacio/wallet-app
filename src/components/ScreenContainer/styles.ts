import styled, { css } from 'styled-components/native';
import { widthPercentageToDP as wp } from 'react-native-responsive-screen';

const outerContainerStyles = css`
  flex: 1;
  background-color: ${({ theme }) => theme.colors.background};
`;

export const ContentView = styled.View`
  flex: 1;
  width: 100%;
  justify-content: center;
  align-items: center;
  padding-horizontal: ${wp('4%')}px;
`;

export const DefaultContainer = styled.View`
  ${outerContainerStyles}
`;

export const ImageContainer = styled.ImageBackground`
  ${outerContainerStyles}
`;
