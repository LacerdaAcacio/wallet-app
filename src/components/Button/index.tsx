import React from 'react';
import { StyledButton, ButtonText } from './styles';
import { TouchableOpacityProps } from 'react-native';

interface CustomButtonProps extends TouchableOpacityProps {
  title: string;
  variant?: 'primary' | 'secondary';
  testID?: string;
}

const Button: React.FC<CustomButtonProps> = ({ title, variant = 'primary', testID, ...rest }) => {
  return (
    <StyledButton variant={variant} testID={testID} {...rest}>
      <ButtonText variant={variant} {...rest}>
        {title}
      </ButtonText>
    </StyledButton>
  );
};

export default Button;
