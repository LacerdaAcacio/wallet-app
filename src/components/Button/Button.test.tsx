import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '../../styles/theme';
import Button from './index';

const renderWithTheme = (component: React.ReactElement) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe('Component: Button', () => {
  it('should apply secondary variant styles when specified', () => {
    const { getByTestId, getByText } = renderWithTheme(
      <Button title="Secondary" variant="secondary" testID="my-button" />,
    );
    const buttonElement = getByTestId('my-button');
    expect(buttonElement).toHaveStyle({
      backgroundColor: theme.colors.secondaryButton,
    });
    const textElement = getByText('Secondary');
    expect(textElement).toHaveStyle({
      color: theme.colors.textOnSecondary,
    });
  });
});
