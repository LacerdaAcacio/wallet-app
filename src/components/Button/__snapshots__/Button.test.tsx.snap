// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Button should render the button with the correct title 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#12C2E9",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "height": 93.5,
      "justifyContent": "center",
      "marginBottom": 33.5,
      "maxHeight": 60,
      "maxWidth": 340,
      "opacity": 1,
      "width": 637.5,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 34,
      }
    }
    variant="primary"
  >
    Click Me
  </Text>
</View>
`;
